xblockutils/studio_editable.py

```diff
@@ -0,0 +1,263 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015 OpenCraft
+# License: AGPLv3
+"""
+This module contains a mixin that allows third aprty XBlocks to be easily edited within edX
+Studio just like the built-in modules. No configuration required, just add
+StudioEditableXBlockMixin to your XBlock.
+"""
+
+# Imports ###########################################################
+
+import json
+import logging
+
+from xblock.core import XBlock
+from xblock.fields import Scope, JSONField, Integer, Float, Boolean, String
+from xblock.exceptions import JsonHandlerError
+from xblock.fragment import Fragment
+from xblock.validation import Validation
+
+from xblockutils.resources import ResourceLoader
+
+# Globals ###########################################################
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)
+
+# Classes ###########################################################
+
+
+class FutureFields(object):
+    """
+    A helper class whose attribute values come from the specified dictionary or fallback object.
+    """
+    def __init__(self, new_fields_dict, newly_removed_fields, fallback_obj):
+        self._new_fields_dict = new_fields_dict
+        self._blacklist = newly_removed_fields
+        self._fallback_obj = fallback_obj
+
+    def __getattr__(self, name):
+        try:
+            return self._new_fields_dict[name]
+        except KeyError:
+            if name in self._blacklist:
+                # Pretend like this field is not actually set, since we're going to be resetting it to default
+                return self._fallback_obj.fields[name].default
+            return getattr(self._fallback_obj, name)
+
+
+class StudioEditableXBlockMixin(object):
+    """
+    An XBlock mixin to provide convenient use of an XBlock in Studio.
```

Comment:
People can already use XBlocks in studio. :) Perhaps you mean convenient editing of XBlocks in Studio?